app/controllers/variable_generation_processes_controller.rb

```diff
@@ -65,4 +67,72 @@ def update
       render :edit
     end
   end
+
+  def save_variable_drafts
+
+    @collection_id = params[:collection_id]
+    generated_variables = JSON.parse(params[:variables_json])
+    num_generated_vars = generated_variables.count
+
+    import_result, drafts_saved = import_variable_drafts(generated_variables)
+
+    if drafts_saved
+      Rails.logger.info "User #{current_user.urs_uid} successfully saved #{generated_variables.count} UMM Variable Generation (UVG) variable records generated from collection #{@collection_id} as Variable Drafts!"
+
+      redirect_to manage_variables_path, flash: { success: "#{num_generated_vars} variable records generated from collection #{@collection_id} saved as Variable Drafts!" }
+    else
+      Rails.logger.error "User #{current_user.urs_uid} encountered an error trying to save #{num_generated_vars} UMM Variable Generation (UVG) generated variables. import_result: #{import_result}"
+
+      @variables = generated_variables
+      @statistics = JSON.parse(params[:statistics])
+
+      flash.now[:error] = "#{num_generated_vars} generated variable records failed to save as Drafts"
+      render :show
+    end
+  end
+
+  private
+
+  def import_variable_drafts(generated_variables)
+    first_generated_var = generated_variables.shift
+
+    mass_import_result = nil
+
+    # transaction will return true if it is successful
+    drafts_saved = VariableDraft.transaction do
+                     first_var = create_single_variable_draft(first_generated_var)
+                     Rails.logger.info "created first variable in transaction:  #{first_var.inspect}"
+                     id_to_start = first_var.id + 1
+
+                     mass_import_result = import_remaining_variable_drafts(generated_variables, id_to_start)
+                     Rails.logger.info "need to save this somehow. mass_import_result: #{mass_import_result.inspect}"
+                   end
+
+    [mass_import_result, drafts_saved]
+  end
+
+  def create_single_variable_draft(generated_var)
```

Comment:
Why does the first variable not have entry id, short name, and native id just curious?   can't these fields be nil so you don't have to have a separate method for the first and all the others?